reference.md

```diff
@@ -1255,14 +1274,18 @@ can use `tostring` to get the name of a symbol.
 
 * `list` - return a list, which is a special kind of table used for code
 * `sym` - turn a string into a symbol
-* `list?` - is the argument a list?
-* `sym?` - is the argument a symbol?
-* `table?` - is the argument a non-list table?
+* `gensym` - generates a unique symbol for use in macros (also available via
+             `<var-name>#`).
```

Comment:
Gensym and autogensym are a bit different, so I think the additional info in the parentheses may be misleading, because `(gensym :foo)` produces `foo_N_`, where N is some unique number but `foo#` produces `foo_N_auto`, and the text may be understood as if the name generated with `gensym` can be available by using this name with `#` appended.

Reply:
Maybe separate them out?

* `gensym` - generates a unique symbol for use in macros, accepts an optional prefix string.
* `<symbol>#` - generates a unique symbol, derived from the given symbol (name?).

Reply:
`<symbol>#` may be confused with REPL representation of some un-printable objects, like `#<function: 0x55937ccb5720>`

I guess it could be

* autogensym - a unique symbol generated when appending `#` to a symbol name inside quoted expression

But I'm open to other ideas